src/vite-preset.ts

```diff
@@ -96,10 +98,14 @@ const buildEndHandler: (config: AwsRemixConfig) => VitePluginConfig['buildEnd']
 
           // workaround dynamic require bug
           // https://github.com/evanw/esbuild/issues/1921#issuecomment-2302290651
-          mainFields: ['module', 'main'],
-          banner: {
-            js: 'import { createRequire } from \'module\'; const require = createRequire(import.meta.url);',
-          },
+          mainFields: isEsm
+            ? ['module', 'main']
+            : undefined,
+          banner: isEsm
+            ? {
+              js: 'import { createRequire } from \'module\'; const require = createRequire(import.meta.url);',
+            }
+            : undefined,
```

Comment:
@brettdh added a check here to only add the workaround when it's an `esm` build, thanks for the feedback! 🙏